File: kyw0716/6주차/튜플.js
function solution(s) {
  let answer = [];
  s = s
    .slice(2, -2)
    .replace(/\,\{/g, "")
    .replace(/\}/g, " ")
    .split(" ")
    .map((v) => v.split(","))
    .sort((a, b) => a.length - b.length);
  answer.push(s[0]);
  for (let i = 1; i < s.length; i++) {
    answer.push(s[i].filter((v) => !s[i - 1].includes(v)));
  }
  return answer.map((v) => Number(v));
}

File: sk000801/LEVEL2/튜플.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        List<Integer> answer = new ArrayList<>();
        String s1 = s.substring(2, s.length());
        String s2 = s1.substring(0, s.length()-4).replace("},{", " ");
        String[] s3 = s2.split(" ");
        Arrays.sort(s3, new Comparator<String>() {
            public int compare (String s1, String s2) {
                return Integer.compare(s1.length(), s2.length());
            }
        });

        for (String ss : s3) {
            String[] numbers = ss.split(",");
            for (int i = 0; i < numbers.length; i++) {
                int num = Integer.parseInt(numbers[i]);
                if (!answer.contains(num)) {
                    answer.add(num);
                }
            }
        }
        int[] realAnswer = new int[answer.size()];
        for (int i = 0; i < answer.size(); i++) {
            realAnswer[i] = answer.get(i);
        }
        return realAnswer;
    }
}